### PR TITLE
Add support for Yes/No/Cancel buttons

### LINKED ISSUE #81: Have more buttons in MessageDialog
It is common to have three buttons in a dialog:
Screenshot 2022-08-11 234333

### PR DESCRIPTION
**Pull request by Adanos020**
Fixes #81 

I based this off of this fork: https://github.com/zphixon/rfd

The Yes/No/Cancel provided by the OS seem to work fine on Linux and macOS, however:
- Didn't compile on Windows,
- There was no support for custom text in the 3 buttons.

I addressed both of these issues and updated the documentation of `MessageDialog` accordingly.

Tested on Windows and Linux (GTK3).

WASM, macOS and Linux (XDG) still need testing. Unfortunately, I don't have easy access to a Mac, so we'll need someone else who does to test this.

### CODE REVIEW

**Comment by Adanos020** [comment]
Opening this again because I accidentally closed #122

File: src/message_dialog.rs [hunk 1]
```diff
@@ -158,16 +154,46 @@ pub enum MessageButtons {
...
+    YesNoCancelCustom(String, String, String),
 }
 
 impl Default for MessageButtons {
     fn default() -> Self {
         Self::Ok
     }
 }
+
+#[derive(Debug, Default)]
```
Comment (PolyMeilex) [inline src/message_dialog.rs:175]:
```suggestion
#[derive(Debug, Default, Clone, PartialEq, Eq)]
```
Just to make it easier to use in if statements without pattern matching

**Review by PolyMeilex — CHANGES_REQUESTED** [review]
Sorry for the delay, this looks great, just one minor comment.

**Comment by PolyMeilex** [comment]
Thanks!

**Pull request merged**